src/gui/Clipboard.cpp

```diff
@@ -52,24 +52,21 @@ void Clipboard::setText(const QString& text, bool clear)
     }
 
     auto* mime = new QMimeData;
-#ifdef Q_OS_MACOS
     mime->setText(text);
+#if defined(Q_OS_MACOS)
     mime->setData("application/x-nspasteboard-concealed-type", text.toUtf8());
-    clipboard->setMimeData(mime, QClipboard::Clipboard);
-#else
-    mime->setText(text);
-#ifdef Q_OS_LINUX
+#elif defined(Q_OS_UNIX)
     mime->setData("x-kde-passwordManagerHint", QByteArrayLiteral("secret"));
-#endif
-#ifdef Q_OS_WIN
+#elif defined(Q_OS_WIN)
     mime->setData("ExcludeClipboardContentFromMonitorProcessing", QByteArrayLiteral("1"));
 #endif
-    clipboard->setMimeData(mime, QClipboard::Clipboard);
 
+#if !defined(Q_OS_MACOS)
```

Comment:
Why was this introduced?

Reply:
This is what the existing code already does:

```
#ifdef Q_OS_MACOS
...
#else
    if (clipboard->supportsSelection()) {
        clipboard->setMimeData(mime, QClipboard::Selection);
    }
#endif
```
https://github.com/keepassxreboot/keepassxc/blob/a6d3f973fa8449f0f7dac864b3bd3928c29c649f/src/gui/Clipboard.cpp#L55-L72

Reply:
Interesting that seems like a bug to me?

Reply:
> Interesting that seems like a bug to me?

I am not at all familiar with clipboard handling on MacOS or Windows, so my PR does not change their functionality.

If this is indeed something to fix, could we do so in another PR?

Reply:
Yes it is a bug, we used to use "setText" now we properly use "setMimeData" so this code should apply to both macOS and other platforms:

![image](https://user-images.githubusercontent.com/2809491/210265812-e5df70f8-4649-4dbd-b4ad-df0f7f5d536a.png)

https://github.com/keepassxreboot/keepassxc/commit/dc7322cc4a08b64e72bf19e3b48502db682a65f6#diff-410a4485502cc323cf953be073436e3326ba2903f4c1912e9872662ce2446687

Reply:
> [dc7322c#diff-410a4485502cc323cf953be073436e3326ba2903f4c1912e9872662ce2446687](https://github.com/keepassxreboot/keepassxc/commit/dc7322cc4a08b64e72bf19e3b48502db682a65f6#diff-410a4485502cc323cf953be073436e3326ba2903f4c1912e9872662ce2446687)

So the newly added `#endif` in line 55 should've probably landed as line 52, between clipboard and selection handling?

The diff you show seems is from 2017, which implies that KeePassXC has been failing to set the selection on MacOS for the last five years?

Looking at it with the context provided, I can add a proper commit fixing this and expand the PR a little.
That should be simplest.

Reply:
> Looking at it with the context provided, I can add a proper commit fixing this and expand the PR a little.

Done with 69e234b.